driver/examples/write_then_read.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+# This example shows how to use CachingCommWriter to write MPI communication data to disk, and then re-run the model using the data from disk.
+
+set -e -x
+
+if [ $# -eq 1 ]; then
+    MPIRUN_CMD=$1
+else
+    MPIRUN_CMD="mpirun -n 6"
+fi
+
```

Comment:
Another way to this kind of stuff I (re)discovered recently is have a default for the `MPIRUN_CMD` env var that get overriden if it was defined in the env prior to the call e.g.:
`MPIRUN_CMD=${MPIRUN_CMD:-mpirun -n 6}"`

That moves the usage of this to a call that look like `MPIRUN_CMD="srun" ./script.sh`

Not really applicable here, but I figured I share, the bash magic being often forgotten :D

Reply:
LGTM. A comment and an ask

Can you do a pass of class docstrings? They are mostly self-explainable but I guess it's never lost to have intent in there. I leave it to you

Can we get a more verbose print on the run? When we were burning node hours for SC, one of the thing that saved us is to spell out the exact experiment we were running early in the run, so we could early out the job in case of mistakes. Mostly namelist stuff & backend. Let me copy here the code

```
        print(
            f"Experiment c{spec.namelist.npx}_{total_ranks}ranks:\n"
            f"  Rank {rank}\n"
            f"  Backend {backend}\n"
            f"  Orchestration: {get_dacemode()}\n"
            f"  N split: {spec.namelist.dynamical_core.n_split}\n"
            f"  K split: {spec.namelist.dynamical_core.k_split}\n"
            f"  dt atmos: {spec.namelist.dynamical_core.dt_atmos}\n"
            f"  Layout: {spec.namelist.layout}\n"
            f"  Grid points: {spec.namelist.npx} - {spec.namelist.npy} - {spec.namelist.npz}\n"
            f"  Grid spacing (Dx,mean): {9000/(spec.namelist.npx-1):.2f}km\n"  # Per FV3 decomposition PDF
            f"  Edge layout for this rank:\n"
            f"    north: {spec.grid.north_edge}\n"
            f"    east: {spec.grid.east_edge}\n"
            f"    south: {spec.grid.south_edge}\n"
            f"    west: {spec.grid.west_edge}\n"
        )
```

Things like which edge seem irrelevant but when looking at inges/dispatch code for the cache it's quite precious.

Reply:
That sounds like a much nicer way to do it, I'll switch to an env var.

Will look at adding some docstrings.

Pretty easy to do even if it's a little out of scope, I'll add that output under a `logging.info` call.

Reply:
Updated.